### ISSUE TITLE
Make retrieval of assets in @podium/client request bound

### ISSUE THREAD

**Issue by trygve-lie** [podium-lib/issues#12]
Currently the methods for getting the assets of a podlet in the `@podium/client` is is not optimal. This is now done by the `.js()` and `.css()` methods on the client class like this:

```js
const Client = require('@podium/client');
const client = new Client();

const podletA = client.register({
    name: 'foo',
    uri: 'http://foo.site.com/manifest.json',
});

const podletB = client.register({
    name: 'foo',
    uri: 'http://bar.site.com/manifest.json',
});

podletA.fetch();
podletB.fetch();

const css = client.css();
const js = client.js();
```

The `.css()` and `.js()` methods will here return the css and js for both podlets. Under the hood it returns the css and js for all podlets which is internally cached.

This causes a race condition when a podlet are in a deploy window. In a deploy window there can be two versions of the same podlet where its random what version the client will hit. In such a window the client will bounce a bit back and forth until the podlet is over on one version. During this bouncing its highly possible that the `.fetch()` method returns the content of a new podlet but the `.css()` and `.js()` methods return the content of a old podlet.

We want the process of getting hold of the css and js request bound.

The `.fetch()` method should change from resolving with just the content of a podlet to resolving with a object where there is one property for the content, one property for css and one property for js.

In other words, this:

```js
const content = await podlet.fetch();
```

should return this:

```js
{
    content: '<section>podlet content</section>',
    css: ['http://foo.site/style.css'],
    js: ['http://foo.site/script.js'],
}
```

The `.stream()` method should emit events for css and js before it starts emitting content data on the stream. Something like this:

```js
const stream = component.stream();
stream.on('css', css => {
    console.log(css);
});
stream.on('js', js => {
    console.log(js);
});
stream.pipe(process.stdout);
```

**NB I:** This change will be breaking.

**NB II:** Current `.js() and `.css()` methods should be kept as is but yeld a deprecation warning for backwards compabillity of asset bundling.

**Comment by digitalsadhu** [comment]
I’ve played around a bit with changing `'content returned from fetch method'` into `{content: 'content returned from fetch method', js: [...], css: [...]}` and backwards compat. Since the typical use case is to call `client.fetch` and pass the result straight to `res.send` and since the new result will be an object instead of a string, then the only way I can see to make this work in a backwards compatible way would be to override `toJSON` rather than `toString` since `res.send` will call `JSON.stringify` on the object. A quick test proved that this approach works. Unsure if this is what we want though. This will ruin any direct calls to `JSON.stringify` on the `result` object. This could be ok in the short term. ie. up until the point that we remove the deprecated string behaviour… I'm learning towards not adding backwards compatibility, any thoughts?

**Comment by digitalsadhu** [comment]
Since we are changing the output of `fetch` from a string to an object. It almost seems a bit weird to just return fallback content under the property `content`.

Eg.
Currently:
```js
const result = await client.fetch(ctx);
// result => '<div>oh no I fell back due to some issue</div>'
```
Will be changed to:
```js
const result = await client.fetch(ctx);
// result => { content: '<div>oh no I fell back due to some issue</div>', js: [], css: [] }
```

I'm wondering if it could be worth indicating that this is a fallback situation. Perhaps one of the following 2 ways:

1. 
```js
{ content: '<div>oh no I fell back due to some issue</div>', fallback: true, js: [], css: [] }
```
or 2.
```js
{ content: '', fallback: '<div>oh no I fell back due to some issue</div>', js: [], css: [] }
```
Thoughts?

**Comment by trygve-lie** [comment]
I am kinda prepared to do a breaking change here. It might be that we want to do a couple of other changes I've been thinking of here. One thing is changing the first parameter to the `.fetch()` method from being the context to take the `HttpIncoming` object:

```js
const result = await client.fetch(HttpIncoming);
```

> I'm wondering if it could be worth indicating that this is a fallback situation. Perhaps one of the following 2 ways:

That might be and its one of the reasons why I would like to change this to returning an object instead of just a string. I would like the returned object to also include http headers from the podlet:

```js
{
    headers: {
        'Access-Control-Allow-Origin': 'maps.google.com',
    },
    content: '<section>podlet content</section>',
    css: ['http://foo.site/style.css'],
    js: ['http://foo.site/script.js'],
}
```
This will be a very needed feature in our "proxy idea" about how to use Podium to migrate off a legacy platform.

So; I am OK by adding such to the object we return but I think this is something we add later. Our priority now is to cater for css/js.

For the two solutions you propose, I would highly prefer the first.
